### PR TITLE
MON-63843-agent-linux-streaming

### PR DESCRIPTION
**Pull request by jean-christophe81**
## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [ ] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/broker-engine/opentelemetry.robot [hunk 1]
```diff
@@ -392,6 +419,381 @@ BEOTEL_SERVE_TELEGRAF_CONFIGURATION_NO_CRYPTED
...
+
+    ${result}    Ctn Check Host Check Status With Timeout    host_1    30    ${start}    0    OK - 127.0.0.1
+    Should Be True    ${result}    hosts table not updated
+
+
+
+BEOTEL_REVERSE_CENTREON_AGENT_CHECK_HOST_CRYPTED
+    [Documentation]    agent check host with encrypted reversed connection and we expect to get it in check result
+    [Tags]    broker    engine    opentelemetry    MON-63843
+    Ctn Config Engine    ${1}    ${2}    ${2}
```
Comment (Evan-Adam) [inline tests/broker-engine/opentelemetry.robot:746]: What is ${1} ${2} ?

File: tests/broker-engine/opentelemetry.robot [hunk 2]
```diff
@@ -2,6 +2,7 @@
 Documentation       Engine/Broker tests on opentelemetry engine server
 
 Resource            ../resources/import.resource
+Library             ../resources/Agent.py
```
Comment (Evan-Adam) [inline tests/broker-engine/opentelemetry.robot:5]: does this should be in import.ressource file to be available everywhere ?

Comment (jean-christophe81) [reply tests/broker-engine/opentelemetry.robot:5]: no it is used only in this file

File: tests/broker-engine/opentelemetry.robot (hunk 1 above)
Comment (jean-christophe81) [reply tests/broker-engine/opentelemetry.robot:746]: 1 2 as numbers, not string

File: tests/resources/resources.resource [hunk 3]
```diff
@@ -384,3 +391,30 @@ Ctn Wait For Engine To Be Ready
...
+        Log To Console    "fail to stop centreon_agent"
+        Ctn Save Logs
+        Ctn Dump Process    centreon_agent    /usr/bin/centagent   centreon_agent
+        Send Signal To Process    SIGKILL    centreon_agent
+        Fail    centreon_agent not correctly stopped (coredump generated)
+    ELSE
+        IF    ${result.rc} != 0
+            Ctn Save Logs
+            Ctn Coredump Info    centreon_agent    /usr/bin/centagent    centreon_agent
+            Should Be Equal As Integers    ${result.rc}    0    centreon_agent not correctly stopped
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply tests/resources/resources.resource:418]: it does compare and fail

File: packaging/centreon-monitoring-agent.yaml [hunk 4]
```diff
@@ -51,6 +51,16 @@ contents:
       owner: centreon-monitoring-agent
       group: centreon-monitoring-agent
 
+overrides:
+  rpm:
+    depends:
+      - openssl-libs  >= 3
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply packaging/centreon-monitoring-agent.yaml]: done

**Review by Evan-Adam — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]

File: agent/src/bireactor.cc [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    return;
+  }
+  to_read = _read_current = std::make_shared<MessageToAgent>();
+  bireactor_class::StartRead(to_read.get());
+}
+
+template <class bireactor_class>
+void bireactor<bireactor_class>::OnReadDone(bool ok) {
+  if (ok) {
+    std::shared_ptr<MessageToAgent> readden;
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply agent/src/bireactor.cc:84]: My taylor is rich

**Review by bouda1 — APPROVED** [review]

**Pull request merged**